### DISCUSSION TITLE
Email parsing

### DISCUSSION THREAD

**Discussion by Sblop** [jonaswinkler/paperless-ng#993] [Feature Requests]
Hi all,

I love paperless. Have used it from the beginning from when Daniel was developing this, and are thrilled about all of the new features.
I do have one feature I miss quite a lot, which I belive whould be a great addition.

Often I receive mails where the invoice is in clear text in the email. - Also it might be essential information written in the email, that is not an attachment itself. In order to save this to paperless usually I print the mail as a pdf and save it to paperless. Quick and easy, however when paperless is parsing the mail anyway for attachments, including all pictures, it might aswell add the mail content. 
I do understand the problem when you have a mail that contains nothing but the attachment and perhaps your signature. It will be annoying to delete this. - However this might be solved by having either 2 email adresses to paperless. One for attachments, and one for mail parsing. - Othervice if I write <parsing> in the subject etc, this could be filtered out.

just my 2 cents.

Have a nice weekend ahead.
Best wishes.
Benjamin

**Comment by kruppjunker** [comment]
Hello Benjamin,
this is a great idea. I got the same problem.
Sometimes i got an email without attachment or any kind of PDF.

A great workflow would be, to copy (or move) my mail into my IMAP Watching Folder.
Then, the mail is gonna be printed as an PDF (maybe virtual PDF printing) and moved into Paperless. When a PDF is also in the attachment, the regulary import should be created.

**Comment by hellspawny2k** [comment]
I would love that feature, too

**Comment by qwertz098** [comment]
I'd appriciate that extension too. In the meanwhile and if you have no privacy concerns, you might use services like https://pdfconvert.me/, what i am doing in appropriate cases. Maybe somebody knows a similar self-hostable project to share with us.

**Comment by smseidl** [comment]
If you are using Docker there is another container you can add to your docker-compose.yaml to do exactly this.  I don't have the name off hand, but it's mentioned in the discussion area or in a pull request.

**Reply by mpfl** [reply]
https://github.com/rob-luke/emails-html-to-pdf

**Comment by mpfl** [comment]
You might be interested in https://github.com/rob-luke/emails-html-to-pdf

It monitors a mailbox via IMAP for email messages without attachments. It turns them into PDFs and then emails them with SMTP.

Unfortunately it doesn't have the capability of depositing items straight into the paperless-ng API, but the developer is intending to implement that feature.

**Reply by torwag** [reply]
I do not get why emailing it back?
If you have a access to the IMAP folder to scan for the mails, you could basically create a new mail with the pdf as attachment and add  it to one of the IMAP folders directly.

**Reply by Sblop** [reply]
Well... I will try this this afternoon. I will keep my email for paperless, and create another one that runs this, and forward the new email/pdf to paperless...
Would it be possible, this emtil to pdf, just saves the pdf directly to a folder on the drive?

**Reply by mpfl** [reply]
> I do not get why emailing it back?

¯\\\_(ツ)\_/¯

> Would it be possible, this emtil to pdf, just saves the pdf directly to a folder on the drive?

Not as far as I can tell.

I'd prefer almost anything than using SMTP to drop it back into the same inbox. I'm running email2pdf on the same machine as paperless so would immensely prefer it to use the API or even drop it into paperless' consumption directory.

**Comment by dwehrmann** [comment]
I'd also like to see this implemented into paperless-ng natively.
In the meantime, I'm going this approach which works very well for my requirements - maybe it helps someone else, too:

- emails without attachments are saved to disk using mail client rules in .eml format
-  .eml file is parsed using mail parser (https://package.wiki/mail-parser) into html format 
- .html file is converted into a PDF using wkhtml (https://wkhtmltopdf.org/libwkhtmltox/)
- .PDF file is consumed by paperless-ng

I have combined all these into a shell script that is run periodically by a cron job.
